p2p/net/udp.go

```diff
@@ -140,33 +220,163 @@ func (n *UDPNet) listenToUDPNetworkMessages(listener net.PacketConn) {
 		copybuf := make([]byte, size)
 		copy(copybuf, buf)
 
-		msg, pk, err := p2pcrypto.ExtractPubkey(copybuf)
+		conn, ok := n.incomingConn[addr.String()]
+		if !ok {
+			n.logger.Debug("Creating new connection ")
+			_, pk, err := p2pcrypto.ExtractPubkey(copybuf)
+
+			if err != nil {
+				n.logger.Warning("error can't extract public key from udp message. (Addr=%v), err=%v", addr.String(), err)
+				continue
+			}
+
+			ns := n.cache.GetOrCreate(pk)
+
+			if ns == nil {
+				n.logger.Warning("coul'd not create session with %v:%v skipping message..", addr.String(), pk.String())
+				continue
+			}
+
+			host, port, err := net.SplitHostPort(addr.String())
+			if err != nil {
+				n.logger.Warning("coudl'nt parse address skipping message from  %v %V", addr.String(), pk)
```

Comment:
```suggestion
				n.logger.Warning("couldn't parse address skipping message from  %v %V", addr.String(), pk)
```

Reply:
also: consider using `%x` for the public key (assuming it's a byte-array)

Reply:
it's a bytearay with a `String` so there's no need.